htdocs/core/actions_addupdatedelete.inc.php

```diff
@@ -366,7 +366,14 @@
 
 // Action validate object
 if ($action == 'confirm_validate' && $confirm == 'yes' && $permissiontoadd) {
-	$result = $object->validate($user);
+	if($object->element == 'inventory'){
+		//Case of the conf INVENTORY_INCLUDE_SUB_WAREHOUSE
+		$include_sub_warehouse = !empty(GETPOST('include_sub_warehouse')) ? GETPOST('include_sub_warehouse') : 0;
```

Comment:
Not sure this is the best way, let's talk about it

Reply:
Move the 
$include_sub_warehouse = !empty(GETPOST('include_sub_warehouse')) ? GETPOST('include_sub_warehouse') : 0;
into the main page and here jsut replace with

`if ($object->element == 'inventory' && !empty($include_sub_warehouse)) {`

Reply:
Done